homeassistant/components/diagnostics/__init__.py

```diff
@@ -138,12 +142,12 @@ async def _async_get_json_file_response(
     hass_sys_info["run_as_root"] = hass_sys_info["user"] == "root"
     del hass_sys_info["user"]
 
-    integration: Integration = hass.data[DATA_INTEGRATIONS][domain]
-
+    integration = await async_get_integration(hass, domain)
     custom_components = {}
     cc_domain: str
     cc_obj: Integration
```

Comment:
I don't think we need these two types, right?

Reply:
Ah indeed, that is not needed anymore. 👍